### PR TITLE
feat/AB#66054_ability_hide_page

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Implemented ability to hide page in applications.

## Ticket

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/66054/

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested verifying if now it's possible to hide a page in applications.

## Sreenshots

Captura de tela de 2023-06-11 15-57-22
Captura de tela de 2023-06-11 15-57-11

# Checklist:

( * == Mandatory ) 

- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by RenzoPrats** [comment]
@AntoineRelief What is missing is just the front-office part that is when get the pages of some application verify if it's visible or not, and if it's not visible the page should be hided from the left sidenav to not be able to select it.

File: apps/back-office/src/app/application/pages/form/form.component.html [hunk 1]
```diff
@@ -4,16 +4,38 @@
...
+          [text]="page ? page.name || '' : step ? step.name || '' : ''"
+          [canEdit]="canEditName"
+          (onChange)="saveName($event)"
+          (formActiveEvent)="formActive = $event"
+        >
+          <h1 class="!m-0 overflow-hidden text-ellipsis">
+            {{ page ? page.name || '' : step ? step.name || '' : '' }}
+          </h1>
+        </safe-editable-text>
+        <button
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/form/form.component.html:18]: i think tailwind is perhaps not merged into this one
we should NOT use material

File: apps/back-office/src/app/application/pages/form/form.component.html [hunk 2]
```diff
@@ -4,16 +4,38 @@
...
+            icon="visibility_off"
+          >
+          </safe-icon>
+        </button>
+        <button
+          mat-icon-button
+          (click)="hidePage()"
+          *ngIf="this.page?.visible"
+        >
+          <safe-icon
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/form/form.component.html:33]: not use of safe-icon but ui-icon

Comment (AntoineRelief) [reply apps/back-office/src/app/application/pages/form/form.component.html:33]: make sure the icon is primary

File: apps/back-office/src/app/application/pages/form/form.component.ts [hunk 3]
```diff
@@ -21,10 +21,14 @@ import {
...
 } from './graphql/queries';
 import {
+  // EditPageVisibilityMutationResponse,
+  // EDIT_PAGE_VISIBILITY,
   EditStepMutationResponse,
   EDIT_STEP,
   EditPageMutationResponse,
   EDIT_PAGE,
+  EditPageVisibilityMutationResponse,
+  EDIT_PAGE_VISIBILITY,
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/form/form.component.ts:31]: useless

File: apps/back-office/src/app/application/pages/form/form.component.ts [hunk 4]
```diff
@@ -320,4 +324,47 @@ export class FormComponent extends SafeUnsubscribeComponent implements OnInit {
       }
     }
   }
+
+  /**
+   * hide a page
+   */
+  hidePage() {
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/form/form.component.ts:331]: incorrect name, should be togglePageVisibility

Comment (AntoineRelief) [reply apps/back-office/src/app/application/pages/form/form.component.ts:331]: + should use editPageMutation, not a new one

File: apps/back-office/src/app/application/pages/form/graphql/mutations.ts [hunk 5]
```diff
@@ -75,3 +75,19 @@ export const EDIT_STEP = gql`
 export interface EditStepMutationResponse {
   editStep: Step;
 }
+
+// === EDIT PAGE VISIBILITY ===
+/** Edit page gql mutation definition */
+export const EDIT_PAGE_VISIBILITY = gql`
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/form/graphql/mutations.ts:81]: to remove

File: apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts [hunk 6]
```diff
@@ -36,10 +36,21 @@ export interface DeleteStepMutationResponse {
...
+  mutation editPage(
+    $id: ID!
+    $name: String
+    $visible: Boolean
+    $permissions: JSON
+  ) {
+    editPage(
+      id: $id
+      name: $name
+      visible: $visible
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts:48]: this one is correct

File: apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts [hunk 7]
```diff
@@ -63,6 +74,22 @@ export interface EditPageMutationResponse {
   editPage: Page;
 }
 
+// === EDIT PAGE VISIBILITY ===
+/** Edit page gql mutation definition */
+export const EDIT_PAGE_VISIBILITY = gql`
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts:79]: to remove

File: apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts [hunk 8]
```diff
@@ -63,6 +74,22 @@ export interface EditPageMutationResponse {
...
+  mutation editPageVisibility($id: ID!, $visible: Boolean) {
+    editPageVisibility(id: $id, visible: $visible) {
+      id
+      visible
+    }
+  }
+`;
+
+/** Edit page gql mutation response interface */
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/graphql/mutations.ts:88]: to remove

File: apps/back-office/src/app/application/pages/workflow/workflow.component.html [hunk 9]
```diff
@@ -11,17 +11,40 @@
...
+          [canEdit]="canEditName"
+          (onChange)="saveName($event)"
+          (formActiveEvent)="formActive = $event"
+        >
+          <h1 class="!m-0 overflow-hidden text-ellipsis">
+            {{ workflow.name }}
+          </h1>
+        </safe-editable-text>
+        <button
+          mat-icon-button
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/workflow.component.html:26]: not material!

Comment (AntoineRelief) [reply apps/back-office/src/app/application/pages/workflow/workflow.component.html:26]: + remove the cdkDragHandle!

File: apps/back-office/src/app/application/pages/workflow/workflow.component.ts [hunk 10]
```diff
@@ -17,6 +17,8 @@ import {
 import {
   EditPageMutationResponse,
   EDIT_PAGE,
+  EditPageVisibilityMutationResponse,
+  EDIT_PAGE_VISIBILITY,
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/workflow.component.ts:21]: to remove

File: apps/back-office/src/app/application/pages/workflow/workflow.component.ts [hunk 11]
```diff
@@ -455,4 +457,48 @@ export class WorkflowComponent
       this.router.navigate(['./'], { relativeTo: this.route });
     }
   }
+
+  /** Hide a page  */
+  hidePage() {
+    const pageId = this.workflow?.page?.id;
+    const visiblePage = this.workflow?.page?.visible ?? true;
+    this.apollo
+      .mutate<EditPageVisibilityMutationResponse>({
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/pages/workflow/workflow.component.ts:466]: this is a duplication of the code, find better ways to do it

**Pull request closed without merging**